client.go

```diff
@@ -0,0 +1,374 @@
+package bloomd
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/fatih/pool.v2"
+	"io"
+	"net"
+	"strings"
+	"time"
+)
+
+/*
+func newConnection(addr *net.TCPAddr, maxAttempts int) (io.ReadWriteCloser, error) {
+	attempted := 0
+
+	var conn net.Conn
+	var err error
+	for attempted < maxAttempts {
+		conn, err = net.DialTCP("tcp", nil, addr)
+		if err == nil {
+			conn.SetReadDeadline(time.Now().Add(time.Second))
+			return conn, nil
+		}
+		attempted++
+	}
+
+	return nil, errors.Wrap(err, "bloomd: unable to establish a connection")
+}
+*/
+
+type client struct {
+	pool        Pool
+	hostname    string
+	timeout     time.Duration
+	maxAttempts int
+	hashKeys    bool
+}
+
+func NewClient(hostname string, hashKeys bool, timeout time.Duration) (Client, error) {
+	return &client{
+		pool:        &singlePool{hostname},
+		hostname:    hostname,
+		timeout:     timeout,
+		maxAttempts: 3,
+		hashKeys:    hashKeys,
+	}, nil
+
+}
+
+func NewPooledClient(hostname string, hashKeys bool, timeout time.Duration, initConns, maxConns int) (Client, error) {
+	p, err := pool.NewChannelPool(initConns, maxConns, func() (net.Conn, error) {
+		return net.Dial("tcp", hostname)
```

Comment:
https://golang.org/pkg/net/#Dial

https://golang.org/pkg/net/#ResolveIPAddr
"Resolving a hostname is not recommended because this returns at most one of its IP addresses."